src/OrchardCore.Modules/OrchardCore.Admin/AdminActionConstraint.cs

```diff
@@ -0,0 +1,29 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.ActionConstraints;
+
+namespace OrchardCore.Admin
+{
+    public class AdminActionConstraint : IActionConstraint
+    {
+        private readonly PathString _adminUrlPrefix;
+
+        public int Order => 0;
+
+        public AdminActionConstraint(PathString adminUrlPrefix)
+        {
+            _adminUrlPrefix = adminUrlPrefix;
+        }
+
+        public bool Accept(ActionConstraintContext context)
+        {
+            if (context.RouteContext.HttpContext.Request.Path.StartsWithSegments(_adminUrlPrefix))
```

Comment:
Does it ensure that there is a trailing slash after the prefix? Should it?

Reply:
No because it’s using PathString and Segments. So always a leading slash never a trailing

Reply:
@sebastienros @deanmarcussen just for infos

Yes, a PathString can't be initialized without a leading slash but it may have a leading slash (that is skipped when we add to it another pathString, not a string) and in that case `.StartsWithSegments()` doesn't work. That's why i created an extension `StartsWithNormalizedSegments()` to overcome this and that we already use in some places.

But anyway we already use adminUrlPrefix as it is to build all our route pattern **strings**. So, maybe good to add a `Trim('/')` at the time we grab the related config value so that e.g we are sure that `/` + `prefix` is okay.